test/codeu/chat/server/JSONTest.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    @Test
+    public void readTest2() {
+        // This function tests if the JSON reader is working if there is a JSON with one type of User in the file
+        Model model = new Model();
+        JSON log = new JSON();
+        Model comparer = new Model();
+        comparer.add(new User(createUuid("1.1074501217"), "Krager", Time.fromMs(12345)));
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input2.json");
+            // The path can vary from device, so this may not compile if it's not the correct path for your project
+            // set up. A suggested path is as follows
+            //model = log.readFromFile("CodeU/test/codeu/chat/server/input2.json");
+        }
+            catch (IOException e) {
+            System.out.println("File not handled properly");
+        }
+        assertEquals(
+               compare(model.userById().all().iterator(), comparer.userById().all().iterator()), true);
+        assertEquals(
+                compare(model.userByText().all().iterator(), comparer.userByText().all().iterator()), true);
+        assertEquals(
+                compare(model.userByTime().all().iterator(), comparer.userByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationById().all().iterator(), comparer.conversationById().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByTime().all().iterator(), comparer.conversationByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByText().all().iterator(), comparer.conversationByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageById().all().iterator(), comparer.messageById().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByText().all().iterator(), comparer.messageByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByTime().all().iterator(), comparer.messageByTime().all().iterator()), true);
+    }
+
+    @Test
+    public void readTest3() {
+        // This function tests if the JSON reader is working if there is a JSON with no data in it
+        JSON log = new JSON();
+        Model model = null;
+        Model comparer = new Model();
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input3.json");
+            // The path can vary from device, so this may not compile if it's not the correct path for your project
+            // set up. A suggested path is as follows
+            //model = log.readFromFile("CodeU/test/codeu/chat/server/input3.json");
+        } catch (IOException e) {
+            System.out.println("File not handled properly");
+        }
+        assertEquals(
```

Comment:
If all of these `assertEquals()` statements are the same in every function, you might extract them into a helper function that takes the `model` and `comparer` as parameters?